sc-kpm/sc-agents-common/agents/ANumberComparison.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+* This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+* Distributed under the MIT License
+* (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+*/
+#include <iostream>
+
+#include <sc-memory/sc_memory.hpp>
+#include <sc-memory/sc_stream.hpp>
+
+#include "ANumberComparison.hpp"
+#include "utils/AgentUtils.hpp"
+#include "utils/CommonUtils.hpp"
+
+
+using namespace std;
+using namespace utils;
+
+namespace scAgentsCommon
+{
+
+SC_AGENT_IMPLEMENTATION(ANumberComparison)
+{
+  if (!edgeAddr.IsValid())
+    return SC_RESULT_ERROR;
+
+  ScAddr question = ms_context->GetEdgeTarget(edgeAddr);
+
+  if (!ms_context->HelperCheckEdge(CoreKeynodes::question, question, ScType::EdgeAccessConstPosPerm))
+  {
+    cout << "This is not a question\n";
```

Comment:
```suggestion
    SC_LOG_DEBUG("This is not a question");
```

Or it is even better to do not log anything there. We are returning error code already